src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -918,6 +920,105 @@ def ingress_present() -> bool:
         warn_message = "WARNING: Removing RGW daemons can cause clients to lose connectivity. "
         return HandleCommandResult(-errno.EBUSY, '', warn_message)
 
+    def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
+        self.mgr.trigger_connect_dashboard_rgw()
```

Comment:
why not call `connect_dashboard_rgw` directly here?

Reply:
it deadlocks.  i didn't look too deeply.

Reply:
actually, it deadlocks in the remove case; this one worked ok in my testing, but doing it async seemed safer.